File: Samples/blazor-add-in/word-blazor-add-in/word-blazor-add-in/Pages/Quotes.razor.js
/* Copyright(c) Maarten van Stam. All rights reserved. Licensed under the MIT License. */
/**
 * Inserts a quote by Emerson into the current cursor location.
 */
export async function insertEmersonQuoteAtSelection() {
    await Word.run(async (context) => {

        // Create a proxy object for the document.
        const thisDocument = context.document;

        // Queue a command to get the current selection.
        // Create a proxy range object for the selection.
        const range = thisDocument.getSelection();

        // Queue a command to replace the selected text.
        range.insertText('"Hitch your wagon to a star."\n', Word.InsertLocation.replace);

        // Synchronize the document state by executing the queued commands,
        // and return a promise to indicate task completion.
        await context.sync();
        console.log('Added a quote from Ralph Waldo Emerson.');
    })
        .catch(function (error) {
            console.log('Error: ' + JSON.stringify(error));
            if (error instanceof OfficeExtension.Error) {
                console.log('Debug info: ' + JSON.stringify(error.debugInfo));
            }
        });
}
/**
 * Inserts a quote by Chekhov at the start of the document body.
 */
export async function insertBruceSchneierQuoteAtTheBeginning() {
    await Word.run(async (context) => {

        // Create a proxy object for the document body.
        const body = context.document.body;

        // Queue a command to insert text at the start of the document body.
        body.insertText('"There is an entire flight simulator hidden in every copy of Microsoft Excel 97."\n', Word.InsertLocation.start);

        // Synchronize the document state by executing the queued commands,
        // and return a promise to indicate task completion.
        await context.sync();
        console.log('Added a quote from Bruce Schneier.');
    })
        .catch(function (error) {
            console.log('Error: ' + JSON.stringify(error));
            if (error instanceof OfficeExtension.Error) {
                console.log('Debug info: ' + JSON.stringify(error.debugInfo));
            }
        });
}
/**
 * Inserts a quote from Steve Ballmer at the end of the document.
 */
export async function insertSteveBallmerAtTheEnd() {
    await Word.run(async (context) => {

        // Create a proxy object for the document body.
        const body = context.document.body;

        // Queue a command to insert text at the end of the document body.
        body.insertText('"Developer, developer, developer!"\n', Word.InsertLocation.end);

        // Synchronize the document state by executing the queued commands,
        // and return a promise to indicate task completion.
        await context.sync();
        console.log('Added a quote from Steve Ballmer.');
    })
        .catch(function (error) {
            console.log('Error: ' + JSON.stringify(error));
            if (error instanceof OfficeExtension.Error) {
                console.log('Debug info: ' + JSON.stringify(error.debugInfo));
            }
        });
}

File: Samples/blazor-add-in/word-blazor-add-in/word-blazor-add-in/Pages/Quotes.razor.cs
/* Copyright(c) Maarten van Stam. All rights reserved. Licensed under the MIT License. */

using Microsoft.AspNetCore.Components;
using Microsoft.JSInterop;

namespace BlazorAddIn.Pages
{
    /// <summary>
    /// Quotes Class
    /// Inserts 'famous' quotes at the current selection, start, or end of the document.
    /// </summary>
    public partial class Quotes
    {
        [Inject]
        public IJSRuntime JSRuntime { get; set; } = default!;
        public IJSObjectReference JSModule { get; set; } = default!;

        protected override async Task OnAfterRenderAsync(bool firstRender)
        {
            if (firstRender)
            {
                JSModule = await JSRuntime.InvokeAsync<IJSObjectReference>("import", "./Pages/Quotes.razor.js");
            }
        }

        /// <summary>
        /// Invokes JavaScript function to insert Emerson quote.
        /// </summary>
        private async Task InsertWaldo() =>
            await JSModule.InvokeVoidAsync("insertEmersonQuoteAtSelection");

        /// <summary>
        /// Invokes JavaScript function to insert Chechov quote.
        /// </summary>
        private async Task InsertBruceSchneier() =>
            await JSModule.InvokeVoidAsync("insertBruceSchneierQuoteAtTheBeginning");

        /// <summary>
        /// Invokes JavaScript function to insert Steve Ballmer quote.
        /// </summary>
        private async Task InsertBallmer() =>
            await JSModule.InvokeVoidAsync("insertSteveBallmerAtTheEnd");
    }
}